hazelcast/src/main/java/com/hazelcast/sql/impl/schema/MappingResolver.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.schema;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+
+/**
+ * Generic interface that resolves mappings based on object name.
+ */
+@FunctionalInterface
+public interface MappingResolver {
+
+    @Nullable
+    Mapping resolve(String name);
+
+    final class CompositeMappingResolver implements MappingResolver {
```

Comment:
This class is only used in test.

Reply:
And in `CalciteSqlOptimizer`.

Reply:
yeah, but there it is used needlessly - a composite resolver composed of a single resolver